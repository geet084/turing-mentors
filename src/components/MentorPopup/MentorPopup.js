import React, { Component } from 'react';
import { Link  } from 'react-router-dom';

export class MentorPopup extends Component {
  state = {
    text: ''
  }

  handleChange = ({ target }) => {
    this.setState({ text: target.value })
  }

  handleSubmit = () => {
    const message = `To user: ${this.props.slack ? this.props.slack : 'anonymous'} -- message: ${this.state.text}`
    this.setState({ text: '' })
    this.props.sendMessage(message)
  }

  render() {
    return (
      <div>
        mentor popup page
        <p>{this.props.name}</p>
        <p>{this.props.location}</p>
        <p>{this.props.cohort}</p>
        <p>{this.props.background}</p>
        <p>{this.props.tech_skills && this.props.tech_skills[0]}</p>
        <div>
          <textarea className="bg-info" name="text" placeholder="send slack message" value={this.state.text} onChange={this.handleChange}></textarea>
        </div>
        <button onClick={this.handleSubmit} >Submit</button>
        <Link to='/mentors'>GO BACK</Link>
      </div>
    )
  }
}

export default MentorPopup;