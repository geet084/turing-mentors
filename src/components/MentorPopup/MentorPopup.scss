.mentor-popup {
  display: flex;
  flex-direction: column;
  justify-content: center;
  .Collapsible {
    margin: 10px auto;
  }
  .links {
    margin: 0;
    a {
      display: unset;
      font-size: .95rem;
      text-decoration: none;
      color: #5f5f5f;
      font-weight: 600;
      &:hover {
        margin-bottom: -3px;
        border-bottom: 3px solid #12f1fc;
        cursor: pointer;
      }
    }
  }
}

.name {
  span {
    margin: auto 7px;
  }
  p {
    margin: 4px 0;
  }
}

.avail {
  display: flex;
  flex-direction: column;
  p {
    margin: 5px;
  }
}

.slack-message .Collapsible__contentInner {
  display: flex;
  flex-direction: column;
}

.contact-info .Collapsible__contentInner {
  display: flex;
  flex-direction: column;
  p {
    margin: 5px;
  }
}

.mentor-popup .Collapsible__contentInner {
  display: flex;
  justify-content: space-around;
  div {
    width: 50%;
  }
  p {
    margin: 5px 0;
  }
}