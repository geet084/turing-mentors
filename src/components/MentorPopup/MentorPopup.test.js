import React from 'react';
import { MentorPopup } from './MentorPopup';
import { shallow } from 'enzyme';

describe('MentorPopup', () => {
  let wrapper;
  let mockProps = {
    id: 1,
    name: 'some name',
    availability: {
      0: [false, true, false],
      1: [false, true, false],
      2: [false, true, false],
      3: [false, true, false],
      4: [false, true, false],
      5: [false, true, false],
      6: [false, true, false]
    },
    tech_skills: ['skill'],
    non_tech_skills: ['skill'],
    contact_details: {slack: 'user'},
    sendMessage: jest.fn(),
  };

  beforeEach(() => {
    wrapper = shallow(<MentorPopup {...mockProps} />);
  });

  it('should match the correct snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should handle change of the slack message input', () => {
    const mockEvent = { target: { value: 'new message' } };

    wrapper.find('.bg-info').simulate('change', mockEvent);

    expect(wrapper.state()).toEqual({ text: 'new message' });
  });

  it('should handle submitting the new slack message', () => {
    const mockEvent = { target: { value: 'new message' } };
    const mockMessage = 'user: new message';

    wrapper.find('.bg-info').simulate('change', mockEvent);
    wrapper.find('button').simulate('click');
    expect(wrapper.state()).toEqual({ text: '' });
    expect(mockProps.sendMessage).toHaveBeenCalledWith(mockMessage);
  });
});