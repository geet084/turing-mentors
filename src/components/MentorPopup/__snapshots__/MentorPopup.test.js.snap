// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MentorPopup should match the correct snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MentorPopup />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "mentor popup page",
        <p />,
        <p />,
        <p />,
        <p />,
        <p />,
        <div>
          <textarea
            className="bg-info"
            name="text"
            onChange={[Function]}
            placeholder="send slack message"
            value=""
          />
        </div>,
        <button
          onClick={[Function]}
        >
          Submit
        </button>,
        <Link
          to="/mentors"
        >
          GO BACK
        </Link>,
      ],
    },
    "ref": null,
    "rendered": Array [
      "mentor popup page",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <textarea
            className="bg-info"
            name="text"
            onChange={[Function]}
            placeholder="send slack message"
            value=""
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "bg-info",
            "name": "text",
            "onChange": [Function],
            "placeholder": "send slack message",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "textarea",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Submit",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Submit",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "GO BACK",
          "to": "/mentors",
        },
        "ref": null,
        "rendered": "GO BACK",
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "mentor popup page",
          <p />,
          <p />,
          <p />,
          <p />,
          <p />,
          <div>
            <textarea
              className="bg-info"
              name="text"
              onChange={[Function]}
              placeholder="send slack message"
              value=""
            />
          </div>,
          <button
            onClick={[Function]}
          >
            Submit
          </button>,
          <Link
            to="/mentors"
          >
            GO BACK
          </Link>,
        ],
      },
      "ref": null,
      "rendered": Array [
        "mentor popup page",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <textarea
              className="bg-info"
              name="text"
              onChange={[Function]}
              placeholder="send slack message"
              value=""
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "bg-info",
              "name": "text",
              "onChange": [Function],
              "placeholder": "send slack message",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "textarea",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Submit",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Submit",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "GO BACK",
            "to": "/mentors",
          },
          "ref": null,
          "rendered": "GO BACK",
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
